tesseract/tesseract_environment/include/tesseract_environment/core/environment.h

```diff
@@ -64,12 +64,64 @@ class Environment
    *
    * Inherited class must call the protected create function
    *
-   * @param scene_graph
-   * @return
+   * @param scene_graph The scene graph to initialize the environment.
+   * @return True if successful, otherwise false
    */
-  virtual bool init(tesseract_scene_graph::SceneGraph::Ptr scene_graph) = 0;
+  template <typename S>
```

Comment:
Should the documentation state what should be passed into this template? ie, a state solver?

Reply:
Also, do you want to do an explicit instantiation of a KDL state solver (maybe you already do).